lib/querystring.js

```diff
@@ -245,7 +244,7 @@ QueryString.parse = QueryString.decode = function(qs, sep, eq, options) {
     if (keys.indexOf(k) === -1) {
       obj[k] = v;
       keys.push(k);
-    } else if (Array.isArray(obj[k])) {
+    } else if (obj[k] instanceof Array) {
```

Comment:
It should be and we're the only one creating the values in this function, so it's safe.